test/functional/feature_maxuploadtarget.py

```diff
@@ -33,48 +37,97 @@ def on_block(self, message):
+        current_size = 0
+        i = 0
+        while size - current_size > MAX_SCRIPT_SIZE and i < len(utxos):
+            spend_addr = node.gettransaction(utxos[i]['txid'])['details'][0]['address']
+            scr = node.getaddressinfo(spend_addr)['scriptPubKey']
+            tx = self.create_tx_with_large_script(int(utxos[i]['txid'], 16), 0, CScript(hex_str_to_bytes(scr)))
+            tx_raw = ToHex(tx)
+            current_size = current_size + len(tx_raw)
+            tx_signed = node.signrawtransactionwithwallet(tx_raw)
+            status = node.sendrawtransaction(tx_signed['hex'], True)
+            i = i + 1
+
+    def create_tx_with_large_script(self, prevtx, n, scriptPubKey):
+        tx = CTransaction()
+        tx.vin.append(CTxIn(COutPoint(prevtx, n), b"", 0xffffffff))
+        tx.vout.append(CTxOut(48*COIN, scriptPubKey))
+        current_size = 0
+        script_output = CScript([b''])
+        while  MAX_SCRIPT_SIZE - current_size  > MAX_SCRIPT_ELEMENT_SIZE:
+            script_output = script_output + CScript([b'\x6a', b'\x51' * (MAX_SCRIPT_ELEMENT_SIZE - 5) ])
+            current_size = current_size + MAX_SCRIPT_ELEMENT_SIZE
+        tx.vout.append(CTxOut(1*COIN, script_output))
+        tx.calc_sha256()
+        return tx
+
     def set_test_params(self):
         self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-maxuploadtarget=800"]]
-
-        # Cache for utxos, as the listunspent may take a long time later in the test
-        self.utxo_cache = []
-
-    def run_test(self):
+        self.num_nodes = 3
         # Before we connect anything, we first set the time on the node
         # to be in the past, otherwise things break because the CNode
         # time counters can't be reset backward after initialization
-        old_time = int(time.time() - 2*60*60*24*7)
-        self.nodes[0].setmocktime(old_time)
+        self.mocktime =  int(time.time() - 2*60*60*24*7)
+        self.genesisBlock =  createTestGenesisBlock(self.signblockpubkey, self.signblockprivkey, self.mocktime)
 
+        self.utxo_cache = []
+
+    def setup_network(self):
+        self.add_nodes(3)
+        self.start_node(0, ["-maxuploadtarget=200"])
+        self.start_node(1)
+        self.start_node(2, ["-maxuploadtarget=800"])
+        connect_nodes_bi(self.nodes, 0,1)
+        connect_nodes_bi(self.nodes, 0,2)
+
+    def run_test(self):
         # Generate some old blocks
         self.nodes[0].generate(31, self.signblockprivkey_wif)
+        self.utxo_cache = self.nodes[0].listunspent()
+        self.sync_all()
 
-        # p2p_conns[0] will only request old blocks
-        # p2p_conns[1] will only request new blocks
-        # p2p_conns[2] will test resetting the counters
+        # p2p_conns[0] will only request old blocks and resetting the counters
+        # p2p_conns[1] will request new and old blocks
+        # p2p_conns[2] will test whitelist
```

Comment:
Why these change necessary?

Reply:
When using peer2 initially, it did not restart.  I guess the problem was timestamp related and needed -reindex. So I just used peer0 to very the resetting of counters. Once the test was passing I didn't backtrack this change.